src/components/NewCommentForm/NewCommentForm.tsx

```diff
@@ -0,0 +1,246 @@
+import React, { useState } from 'react';
+import classNames from 'classnames';
+
+import { Post } from '../../types/Post';
+import { Comment } from '../../types/Comment';
+
+import { addComments } from '../../services/comments';
+
+type Props = {
+  selectedPost: Post;
+  setComments: (value: Comment[] | { (prev: Comment[]): Comment[] }) => void,
```

Comment:
let's make it shorter
look [here](https://stackoverflow.com/questions/65823778/how-can-i-define-type-for-setstate-when-react-dispatchreact-setstateactionstri)